### DISCUSSION TITLE
Strange ECONNRESET errors

### DISCUSSION THREAD

**Discussion by wolflu05** [SchoolUtils/WebUntis#92] [Q&A]
Hello, I know this problem is most likely (maybe useragent) not related to this code, but to webuntis. I want to ask for your experience. Starting today morning I always get the following error: 
```
Error: read ECONNRESET
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20
```

I think this could be related to some protection mechanisms announced on https://status.webuntis.com due to DDos attacks. Strange is, that if I try to do this request exactly the same in postman, sometimes it works and sometimes not.
Do you also experience this error?

**Comment by TheNoim** [comment]
Does it work again? I am pretty sure, this was because of the DDoS attack.

**Reply by wolflu05** [reply]
Yes, it was working again on the next morning. I was just asking for the other users experience. (I never had this in the last 3/4 year since I'm using this in production). 

My idea was just that maybe the useragent or some headers are more likely triggers for the spam protection from webuntis. But I couldn't find something. I pasted the exact request into postman and 1/10 requests succeed. It was more like random, but if one succeed he next couple of requests also succeeded, but then it stopped working and the connection was closed. So yeah maybe a protection mechanism.

But now everything is working again. Had no issues since then.

**Reply by wolflu05** [reply]
Today same error since 7:06am. Must be related to the protection mechanisms, very annoying, but we can't change something. ![image](https://user-images.githubusercontent.com/76838159/208902452-4b025d0b-39bf-4ebf-a92b-da8f0008996f.png)

**Reply by vascYT** [reply]
You guys still have the issue, right?

**Reply by wolflu05** [reply]
@vascYT No idea, shut the thing down because we have holidays now. But I did a few requests with postman. The first 4 failed with the econnreset error, but the next 20 were successful.

**Reply by vascYT** [reply]
Okay thanks :( Well let's hope it works after the holidays.

**Reply by vascYT** [reply]
It seems to be working again. Let's hope it stays like that 🤞
